src/cmds/scripts/pbs_habitat.in

```diff
@@ -318,15 +318,31 @@ if [ "${PBS_START_SERVER:-0}" != 0 ] ; then
 			exit $ret
 		fi
 		server_started=1
+		
+		PSI_VAL=
+		# Check if PBS_SERVER_INSTANCES is set for multi-server
+		if [ ! -z "${PBS_SERVER_INSTANCES}" ]; then
+			# We want to set PBS_SERVER_INSTANCES to just the server that's being started
+			# this is needed so that qmgr doesn't try to contact other servers, which may not be up yet
+			IFS=',' read -ra PSIARR <<< "$PBS_SERVER_INSTANCES"
+			for psi_svr in "${PSIARR[@]}"; do
+				psi_hostport=(${psi_svr//:/ })
+				psi_port=${psi_hostport[1]}
+				if [[ "$psi_port" == "$PBS_BATCH_SERVICE_PORT" ]]; then
+					PSI_VAL=$psi_svr
+					break
+				fi
+			done
+		fi
 		if is_cray_xt; then
-			${PBS_EXEC}/bin/qmgr <<-EOF > /dev/null
+			PBS_SERVER_INSTANCES=$PSI_VAL ${PBS_EXEC}/bin/qmgr <<-EOF > /dev/null
```

Comment:
Why not set at one place rather than setting in front of individual commands?

There are other places you might want to do the same. for instance, pbsd_init calls qterm and pbs_dataservice calls qstat during start/stop of pbs.